### PR TITLE
[machine learning] update pytorch user guide with pytorch 2.0 improvements

### PR DESCRIPTION
**Pull request by snadampal**
*Issue #, if available:*

*Description of changes:*
This PR updates the pytorch user guide with the upcoming 2.0 release improvements, adds torch bench benchmark details, and cleans up the outdated code snippets.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by AGSaidi — APPROVED** [review]

File: machinelearning/pytorch.md [hunk 1]
```diff
@@ -5,54 +5,73 @@
...
 
 # How to use PyTorch on Graviton CPUs
-There are multiple levels of software package abstractions available: Python wheel (easiest option but multiple wheels to be installed), AWS DLC (Deep Learning Container, comes with all the packages installed) and the Docker hub images (additionally comes with MLPerf benchmarks). Examples of using each method are below.
+There are multiple levels of software package abstractions available: Python wheel (easiest option to get the release features but multiple wheels to be installed), AWS DLC (Deep Learning Container, comes with additional optimizations on top of wheels, and all the packages installed) and the Docker hub images (additionally comes with downstream experimental features). Examples of using each method are below.
```
Comment (AGSaidi) [inline machinelearning/pytorch.md:8]: Lead with DLC and say best perf?

File: machinelearning/pytorch.md [hunk 2]
```diff
@@ -5,54 +5,73 @@
...
-pip install torchtext
+# Install PyTorch and extensions
+python3 -m pip install torch
+python3 -m pip install torchvision torchaudio torchtext
 ```
 
 **AWS Graviton PyTorch DLC**
 
-3Q'22 DLCs are PyTorch1.12 based but include the optimizations that were included in PyTorch1.13, so the performance is nearly identical.
+1Q'23 AWS Graviton DLCs are based on PyTorch2.0, but they also include additional optimizations from PyTorch master. The DLCs improve the PyTorch inference performance on Graviton by 2x to 4x compared to the previous releases, including up to 4x improvement for Resnet50 and up to 3x for Bert, making Graviton3 the most cost effective CPU platform on the AWS cloud for these models.
```
Comment (AGSaidi) [inline machinelearning/pytorch.md:28]: Another word for master pleas

Comment (AGSaidi) [reply machinelearning/pytorch.md:28]: The text is duplicated here ans above roughly.

File: machinelearning/pytorch.md [hunk 3]
```diff
@@ -150,32 +149,87 @@ If the tensor ops and shapes are still not executed with ACL gemm kernels, pleas
...
+5. The above triaging steps cover typical issues due to the missing runtime configurations. If you are stuck with any of these steps or if the performance is still not meeting the target, please raise an issue on [aws-graviton-getting-started](https://github.com/aws/aws-graviton-getting-started) github.
 
 # Building PyTorch from source
 
-While the packages for docker container and wheels provide stable baseline for ML application development and production, they lack the latest fixes and optimizations from the master branch. The scripts for building the torch, torch-xla and torchvision wheels are avaiable in torch-xla repo. This section provides instructions for using those scripts.
+While the packages for deep learning containers and wheels provide stable baseline for ML application development and production, they lack the latest fixes and optimizations from the master branch. This section provides instructions for building the torch wheels from sources.
```
Comment (AGSaidi) [inline machinelearning/pytorch.md:175]: Do we want ppl to do this?

File: machinelearning/pytorch.md (hunk 1 above)
Comment (snadampal) [reply machinelearning/pytorch.md]: Hi @AGSaidi , addressed all your feedback in the next update

**Comment by RamaMalladiAWS** [comment]
Can be merged?

**Pull request merged**